### PR TITLE
Fix eslint issues and turn on prettier by default

### PR DESCRIPTION
**Pull request by grabbou**
A new take on `prettier` and `eslint`.

Addresses previous comments by eliminating precommit hook in favour of eslint integrated approach.

I believe this can be merged as everyone accepted previous approach except the `precommit` that is now gone.

I removed `website` and `examples` from the diff.

CC: @mjackson - you were interested to see how it can be set up

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,8 +27,10 @@
     "start": "node ./node_modules/react-native/local-cli/cli.js start --config ./rn-cli.config.js",
     "run-playground-ios": "cd examples/NavigationPlayground && react-native run-ios",
     "run-playground-android": "cd examples/NavigationPlayground && react-native run-android",
-    "test": "jest",
+    "test": "npm run lint && npm run flow && npm run jest",
+    "jest": "jest",
     "lint": "eslint src",
+    "format": "eslint --fix src",
```
Comment (satya164) [inline package.json]: Can we run prettier along with eslint too?

**Comment by grabbou** [comment]
Yeah, we use eslint prettier plugin which has an auto fixable rule. That
means eslint fix also runs prettier behind the scenes.

File: package.json (hunk 1 above)
Comment (AlanFoster) [inline package.json]: There's a few open source plugins to help with this; From the prettier [readme](https://github.com/prettier/prettier#related-projects)

When I integrated one of the wrapping eslint prettier plugins, anything that wasn't prettied correctly would be raised as an issue. That integration also helps nuke the custom `format` script, as it just boils down to `npm run lint -- --fix`

Comment (grabbou) [inline package.json]: See my comment below :) we set it up in the way that eslint fix also runs prettier. Pluigns are set in a way that they don't conflict with each other.

**Comment by nihgwu** [comment]
OMG, this breaks all the PRs again 😂

**Comment by grabbou** [comment]
This is just a style change, in most cases it's a clean merge.

**Comment by nihgwu** [comment]
I'm updating my PRs

**Comment by grabbou** [comment]
Cool, I just did that for one of them that I am planning to merge.

**Comment by nihgwu** [comment]
I'll leave that one to you 😄

**Pull request merged**